### PR TITLE
Feature/game over without crashing

### PR DESCRIPTION
**Pull request by lilianaweimer**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type
 
 
Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe): 

## What is the current behavior?

- The game was playable but would crash at the end because it was trying to render questions when there were none

## What is the new behavior?

- The game is now completely playable!
- The game over component renders with the score and/or lives left
- On clicking "home" on the game over component, the game resets and loads the form

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Hooray for us! We made a playable trivia game! 🤠

### CODE REVIEW

File: src/reducers/setQuestions.js [hunk 1]
```diff
@@ -16,10 +16,14 @@ export const hasErrored = (state = '', action) => {
   }
 }
 
-export const setQuestions = (state = [], action) => {
+const initialState = [];
```
Comment (Billwilke42) [inline src/reducers/setQuestions.js:19]: Nic work on these resets

File: src/GameOver/GameOver.js [hunk 2]
```diff
@@ -1,14 +1,30 @@
-import React from 'react'
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { reset } from '../actions';
 
 const GameOver = (props) => {
```
Comment (Billwilke42) [inline src/GameOver/GameOver.js:7]: I like we made this its own page

File: src/Form/Form.js [hunk 3]
```diff
@@ -39,11 +38,11 @@ class Form extends React.Component {
...
-    await this.props.getQuestions(this.state.round2, 'easy')
-    await this.props.getQuestions(this.state.round3, 'medium')
-    await this.props.getQuestions(this.state.round4, 'medium')
-    await this.props.getQuestions(this.state.round5, 'hard')
-    await this.props.getQuestions(this.state.round6, 'hard')
+    // await this.props.getQuestions(this.state.round2, 'easy')
+    // await this.props.getQuestions(this.state.round3, 'medium')
+    // await this.props.getQuestions(this.state.round4, 'medium')
+    // await this.props.getQuestions(this.state.round5, 'hard')
+    // await this.props.getQuestions(this.state.round6, 'hard')
```
Comment (KyleWong2510) [inline src/Form/Form.js:45]: comment these back in before turn in

File: src/Chalkboard/Chalkboard.js [hunk 4]
```diff
@@ -13,19 +13,24 @@ import {
...
       question: this.props.questions[this.props.currentRound][this.props.currentQuestion],
       isCorrect: null,
-      gameOver: false
+      gameOver: false,
+      answerCount: 0,
     }
   }
   
   checkAnswer = (e) => {
+    this.setState((prevState) => {
```
Comment (Billwilke42) [inline src/Chalkboard/Chalkboard.js:31]: Nice, work on getting the game to reset at the end

File: src/actions/index.js [hunk 5]
```diff
@@ -52,4 +52,8 @@ export const isLoading = bool => ({
 export const hasErrored = message => ({
   type: 'HAS_ERRORED',
   message
+})
+
+export const reset = () => ({
+  type: 'RESET'
```
Comment (KyleWong2510) [inline src/actions/index.js:58]: Glad we can make this work throughout all reducers

**Pull request merged**